manifests/0000_70_cluster-network-operator_01_egr_crd.yaml

```diff
@@ -0,0 +1,186 @@
+    controller-gen.kubebuilder.io/version: v0.4.0
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+  creationTimestamp: null
+  name: egressrouters.network.operator.openshift.io
+spec:
+  group: network.operator.openshift.io
+  names:
+    kind: EgressRouter
+    listKind: EgressRouterList
+    plural: egressrouters
+    singular: egressrouter
+  scope: Namespaced
+  versions:
+  - additionalPrinterColumns:
+    - jsonPath: .status.conditions[*].type
+      name: Condition
+      type: string
+    - jsonPath: .status.conditions[*].status
+      name: Status
+      type: string
+    name: v1
+    schema:
+      openAPIV3Schema:
+        description: EgressRouter is a single egressrouter pod configuration object.
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: EgressRouterSpec contains the configuration for an egress router.
+            properties:
+              addresses:
+                description: List of IP addresses to configure on the pod's secondary interface.
+                items:
+                  description: EgressRouterAddress contains a pair of IP CIDR and gateway to be configured on the router's interface
+                  properties:
+                    gateway:
+                      description: IP address of the next-hop gateway, if it cannot be automatically determined. Can be IPv4 or IPv6.
+                      type: string
+                    ip:
+                      description: IP is the address to configure on the router's interface. Can be IPv4 or IPv6.
+                      type: string
+                  required:
+                  - ip
+                  type: object
+                type: array
+              modeConfig:
+                description: ModeConfig represents router's mode configuration.
+                properties:
+                  allowedDestinationsRedirect:
+                    description: List of destinations that the pod is allowed to connect to in redirect mode. If unspecified, the router will not be configured to connect to specific remote destination. Therefore if the mode is "Redirect" then this field must be set.
```

Comment:
Q: I wonder what happens if the destination is not set. Need to test how this is handled today, from looking at the code, it looks like it will error out: https://github.com/openshift/egress-router-cni/blob/dadd5efd4ea9c10788feabb04392e868d126a8a6/pkg/macvlan/macvlan.go#L377

Reply:
We might need to add this logic to admission controller, because I am not sure how to handle the relational assignment in kubebuilder.

Reply:
update: might need to try one of the `kubebuilder:validation:MinItems`/ `kubebuilder:validation:ExclusiveMinimum` or something. I remember trying something like this for the macvlan config... but couldn't find the right validator...

Reply:
ok so it is https://github.com/openshift/cluster-network-operator/pull/125/files/c15bf3c625a7a7833774696f85f2d1531cc39faa#r269165577 #709 
`minProperties` and `maxProperties`. For some weird reason they don't get detected in the autogeneration. Had to fix it manually. But they work as expected.

`The EgressRouter "egress-router-1115" is invalid: spec.modeConfig: Invalid value: 2: spec.modeConfig in body should have at least 2 properties`